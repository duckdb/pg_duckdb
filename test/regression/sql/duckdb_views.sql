CREATE USER duckdb_view_admin IN ROLE duckdb_group;
CREATE USER duckdb_view_user1;

SET ROLE duckdb_view_admin;

CREATE VIEW duckdb_view AS SELECT * FROM read_parquet('/nonexisting.parquet');

SET ROLE duckdb_view_user1;

SELECT * from duckdb_view LIMIT 1;

SET ROLE duckdb_view_admin;
GRANT SELECT ON duckdb_view TO duckdb_view_user1;
SET ROLE duckdb_view_user1;

SELECT * from duckdb_view LIMIT 1;
