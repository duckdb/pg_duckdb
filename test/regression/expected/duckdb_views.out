CREATE USER duckdb_view_admin IN ROLE duckdb_group;
SET ROLE duckdb_view_admin;
CREATE USER duckdb_view_user1;
CREATE VIEW duckdb_view AS SELECT * FROM read_parquet('/nonexisting.parquet');
SET ROLE duckdb_view_user1;
SELECT * from duckdb_view LIMIT 1;
ERROR:  permission denied for view duckdb_view
SET ROLE duckdb_view_admin;
GRANT SELECT ON duckdb_view TO duckdb_view_user1;
SET ROLE duckdb_view_user1;
SELECT * from duckdb_view LIMIT 1;
ERROR:  DuckDB execution is not allowed because you have not been granted the duckdb.postgres_role
